proof/invariant-abstract/DetSchedAux_AI.thy

```diff
@@ -155,6 +154,15 @@ lemma invoke_untyped_etcb_at:
                     hoare_drop_impE_E)
   done
 
+lemma invoke_untyped_etcb_at:
+  "\<lbrace>\<lambda>s::'state_ext state. etcb_at P t s \<and> invs s \<and> st_tcb_at (Not o inactive and Not \<circ> idle) t s \<and> ct_active s \<and> valid_untyped_inv ui s\<rbrace>
+   invoke_untyped ui
+   \<lbrace>\<lambda>_. etcb_at P t\<rbrace>"
+  apply (rule hoare_post_imp[where Q'="\<lambda>_ s. st_tcb_at (Not \<circ> inactive) t s \<and> (st_tcb_at (Not \<circ> inactive) t s \<longrightarrow> etcb_at P t s)"])
```

Comment:
Does it not make sense to clobber `invoke_untyped_etcb_at`?

Reply:
Do you mean get rid of the original lemma? If so, the proofs that were using it are pretty fiddly and expect the implication in the postcondition, while I'm using it in more standard ways that don't.